### ISSUE TITLE
Issue with Gradle on JDK 9

### ISSUE THREAD

**Issue by galuszkak** [keeganwitt/docker-gradle#27]
I have the same build.gradle file but one I've use with `gradle:latest` and second one with `gradle:4.2.0-jdk9`. What is my problem, that with second one I've got an error:

```
java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
```

I suspect this has to do something with ca-certificates/ca-certificates-java package.
This is full error stacktrace:
```> Could not resolve org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M3.
     Required by:
         project :
      > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M3.
         > Could not get resource 'https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
            > Could not GET 'https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M3.
         > Could not get resource 'https://repo.spring.io/snapshot/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
            > Could not GET 'https://repo.spring.io/snapshot/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M3.
         > Could not get resource 'https://repo.spring.io/milestone/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
            > Could not GET 'https://repo.spring.io/milestone/org/springframework/boot/spring-boot-gradle-plugin/2.0.0.M3/spring-boot-gradle-plugin-2.0.0.M3.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
   > Could not resolve org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.51.
     Required by:
         project :
      > Could not resolve org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.51.
         > Could not get resource 'https://repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
            > Could not GET 'https://repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.51.
         > Could not get resource 'https://repo.spring.io/snapshot/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
            > Could not GET 'https://repo.spring.io/snapshot/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.51.
         > Could not get resource 'https://repo.spring.io/milestone/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
            > Could not GET 'https://repo.spring.io/milestone/org/jetbrains/kotlin/kotlin-gradle-plugin/1.1.51/kotlin-gradle-plugin-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
   > Could not resolve org.jetbrains.kotlin:kotlin-allopen:1.1.51.
     Required by:
         project :
      > Could not resolve org.jetbrains.kotlin:kotlin-allopen:1.1.51.
         > Could not get resource 'https://repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
            > Could not GET 'https://repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.jetbrains.kotlin:kotlin-allopen:1.1.51.
         > Could not get resource 'https://repo.spring.io/snapshot/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
            > Could not GET 'https://repo.spring.io/snapshot/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve org.jetbrains.kotlin:kotlin-allopen:1.1.51.
         > Could not get resource 'https://repo.spring.io/milestone/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
            > Could not GET 'https://repo.spring.io/milestone/org/jetbrains/kotlin/kotlin-allopen/1.1.51/kotlin-allopen-1.1.51.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
```

**Comment by keeganwitt** [comment]
This is a known issue.  I've opened a defect against the openjdk Docker image (https://github.com/docker-library/openjdk/issues/145).  As soon as it's fixed, I'll make sure to publish updated images with the fix.

**Comment by keeganwitt** [comment]
[8189357](http://bugs.java.com/view_bug.do?bug_id=JDK-8189357) was the cause of docker-library/openjdk#145, and since Oracle has taken forever to fix this (and it looks like the fix is related to [8189131](http://bugs.java.com/view_bug.do?bug_id=JDK-8189131) which doesn't look like it's gonna land until 9.0.4), I went ahead and added a workaround.
